File: Task3/ex3Ruby/palindrom.rb
# Проверяет, является ли число палиндромом (формат ABC == CBA)
def is_palindrome(num)
  num = num.abs
  return false unless num >= 100 && num <= 999  # Только 3-значные
  (num / 100) == (num % 10)  # Сравниваем первую и последнюю цифру
end

# Основной код
print "Введите числа через пробел: "
input = gets.chomp  # Получаем ввод

print "Результат: "
first = true  # Флаг первого элемента

input.split.each do |token|
  next unless token =~ /^-?\d+$/  # Пропускаем не числа
  
  num = token.to_i
  if num.abs.between?(100, 999)  # Проверяем 3-значность
    print ", " unless first  # Разделитель для элементов
    print "#{num} - #{is_palindrome(num) ? 'YES' : 'NO'}"
    first = false
  end
end

puts first ? "нет трехзначных чисел" : ""  # Сообщение если чисел нет

File: Task3/ex3Cpp/palindrom.cpp
#include <iostream>

// Проверка, является ли число трехзначным палиндромом
bool isPalindrome(int num) {
    num = abs(num); // Работаем с модулем числа
    if (num < 100 || num > 999) return false;
    return (num / 100) == (num % 10); // Сравниваем первую и последнюю цифру
}

int main() {
    std::cout << "Введите числа через пробел: ";
    int num;
    
    // Обработка каждого числа из ввода
    while (std::cin >> num) {
        if (abs(num) >= 100 && abs(num) <= 999) { // Проверка на трехзначность
            std::cout << num << " - " << (isPalindrome(num) ? "YES" : "NO") << "\n";
        }
    }
    
    return 0;
}